layers/+window-management/spacemacs-layouts/funcs.el

```diff
@@ -117,6 +122,25 @@ perspectives does."
                       (projectile-switch-project-by-name project)))))))
    :buffer "*Helm Projectile Layouts*"))
 
+(defun spacemacs/helm-persp-switch-open-project (arg)
+  (interactive "P")
+  (helm
+   :sources
+   (helm-build-in-buffer-source "*Helm Switch Project Layout*"
+     :data (lambda ()
+             (if (projectile-project-p)
+                 layouts-projectile
+               projectile-known-projects))
```

Comment:
I'll work on the freezing thing. 

I didn't add the known project as this is only to switch between already open ones. To see the known projects you have `SPC p l`.